pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -129,6 +130,19 @@ func (rc *reconciler) syncStates() {
 	}
 }
 
+func (rc *reconciler) isExclusiveVolume(volumeSpec *volume.Spec) bool {
+	if volumeSpec.PersistentVolume == nil {
+		return false
+	}
+	// check if this volume can be attached to multiple PODs/nodes, if yes, return false
+	for _, accessMode := range volumeSpec.PersistentVolume.Spec.AccessModes {
```

Comment:
NO, we cannot use access mode from the PV here. We have to use readOnly from e.g.  spec.Volume.GCEPersistentDisk.ReadOnly. Depending on which volume plugin it is, we have to go though different plugin such as spec.Volume.AWSElasticBlockStore.ReadOnly. 
In pod yaml file, user can specify volume directly without involving PV at all. Even if you are using PV, the volume attachment mode is still determined by ReadOnly option specified in volume spec. 

apiVersion: v1
kind: Pod
metadata:
  name: jing-volume-pod 
  labels:
    name: mypod
spec:
  containers:
  - name: hyper-pod
    image: redis
    volumeMounts:
    - name: jing-foo2
      mountPath: /test-pd
  volumes:
  - name: jing-foo2
    gcePersistentDisk:
      pdName: aws-disk
      fsType: ext4
      readOnly: true